### ISSUE TITLE
ext-curl missing error even when apt-get install says "php5-curl is already the newest version"

### ISSUE THREAD

**Issue by therobyouknow** [guzzle/guzzle#1586]
Following running a composer install or composer update, I get: `" guzzle/guzzle v3.0.7 requires ext-curl * -> the requested PHP extension curl is missing from your system."`

Searching on the web, several who've had this problem found it to be solved by: `sudo apt-get install php5-curl` but unfortunately this has not resolved it for me. 

My setup:
Ubuntu 14.04 LTS 64bit VM provisioned using Vagrant puphpet.com .zip downloaded today
includes php 5.6, running on MacOS El Capitan Host.

Output showing error, followed by evidence of latest version of php5-curl and composer.json contents.

Is this issue appropriate to raise here? Thank you. If I find a solution before anyone else does, I will be sure to post it here.

```
$ composer update
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - guzzle/guzzle v3.0.7 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - guzzle/guzzle v3.0.6 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - guzzle/guzzle v3.0.5 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - guzzle/guzzle v3.0.4 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - guzzle/guzzle v3.0.3 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - guzzle/guzzle v3.0.2 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - guzzle/guzzle v3.0.1 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - guzzle/guzzle v3.0.0 requires ext-curl * -> the requested PHP extension curl is missing from your system.
    - Installation request for guzzle/guzzle 3.0.* -> satisfiable by guzzle/guzzle[v3.0.0, v3.0.1, v3.0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.0.7].

  To enable extensions, verify that they are enabled in those .ini files:
    - /etc/php/5.6/cli/php.ini
    - /etc/php/5.6/cli/conf.d/10-mysqlnd.ini
    - /etc/php/5.6/cli/conf.d/10-opcache.ini
    - /etc/php/5.6/cli/conf.d/10-pdo.ini
    - /etc/php/5.6/cli/conf.d/20-calendar.ini
    - /etc/php/5.6/cli/conf.d/20-ctype.ini
    - /etc/php/5.6/cli/conf.d/20-exif.ini
    - /etc/php/5.6/cli/conf.d/20-fileinfo.ini
    - /etc/php/5.6/cli/conf.d/20-ftp.ini
    - /etc/php/5.6/cli/conf.d/20-gettext.ini
    - /etc/php/5.6/cli/conf.d/20-iconv.ini
    - /etc/php/5.6/cli/conf.d/20-intl.ini
    - /etc/php/5.6/cli/conf.d/20-json.ini
    - /etc/php/5.6/cli/conf.d/20-mysql.ini
    - /etc/php/5.6/cli/conf.d/20-mysqli.ini
    - /etc/php/5.6/cli/conf.d/20-pdo_mysql.ini
    - /etc/php/5.6/cli/conf.d/20-phar.ini
    - /etc/php/5.6/cli/conf.d/20-posix.ini
    - /etc/php/5.6/cli/conf.d/20-readline.ini
    - /etc/php/5.6/cli/conf.d/20-shmop.ini
    - /etc/php/5.6/cli/conf.d/20-sockets.ini
    - /etc/php/5.6/cli/conf.d/20-sysvmsg.ini
    - /etc/php/5.6/cli/conf.d/20-sysvsem.ini
    - /etc/php/5.6/cli/conf.d/20-sysvshm.ini
    - /etc/php/5.6/cli/conf.d/20-tokenizer.ini
    - /etc/php/5.6/cli/conf.d/zzzz_custom.ini
  You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.

[03:31 PM]-[vagrant@jira61]-[/var/www/atlassian-oauth-examples/php] 
$ sudo apt-get install php5-curl
Reading package lists... Done
Building dependency tree       
Reading state information... Done
php5-curl is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 113 not upgraded.

[03:32 PM]-[vagrant@jira61]-[/var/www/atlassian-oauth-examples/php] 
$ 

```

here is the composer.json

```
$ cat composer.json 
{
    "name": "stanlemon/atlassian-oauth",
    "description": "Atlassian PHP OAuth Example",
    "keywords": ["atlassian", "jira", "oauth", "guzzle"],
    "license": "MIT",
    "autoload": {
        "psr-0": { 
            "": "src/"
         }
    },
    "minimum-stability": "dev",
    "require": {
        "guzzle/guzzle": "3.0.*",
        "silex/silex": "1.0.*",
        "twig/twig": "1.*"
    }
}

[03:32 PM]-[vagrant@jira61]-[/var/www/atlassian-oauth-examples/php] 
$ 

```

**Comment by Nyholm** [comment]
What does `php -i | grep curl` say? I believe that you haven't enabled the extension in your php.ini.

**Comment by therobyouknow** [comment]
Ny Nyholm. Yes and the curl.so for php 5.6 is not available either. I'll raise a ticket on puphpet.com about that.

**Comment by therobyouknow** [comment]
The root cause was that the curl.so was missing and not referred from the php.ini (i.e. enabled as you say). The curl.so was missing because my puphpet.com-based configuration for building the Ubuntu guest VM using Vagrant did not use curl - my mistake I had not included it. All of these can be fixed as follows.

Solution steps:
1) In config.yaml for puphpet.com-based configuration for vagrant, ensure curl is listed:

```
php:
    install: '1'
    settings:
        version: '56'
    modules:
        php:
            - cli
            - intl
            - curl
        pear: {  }
        pecl: {  }
```

Then for this change (to include curl) to take effect either do `vagrant provision` on an existing VM or start afresh by `vagrant destroy` then `vagrant up` (I started afresh, I didn't test the `vagrant provision` on an existing VM for this particular change by I expect this to work)

This resulted in the curl.so to be created for the php version I chose (also in the config.yaml), 5.6 - i.e. on Ubuntu 14.04 LTS 64bit, the location of the curl.so file is here: /usr/lib/php/20131226/curl.so

So in the php.ini file, I enabled curl by adding the line:
`extension=/usr/lib/php/20131226/curl.so`
in the [curl] section, so it looks like:

```
[curl]
; A default value for the CURLOPT_CAINFO option. This is required to be an
; absolute path.
;curl.cainfo =
extension=/usr/lib/php/20131226/curl.so
```

Then when I ran composer install in the source folder, the error I had does not now occur, the install proceeds beyond guzzle.

A nice guide for reference about curl.so and it's location etc is here: http://stackoverflow.com/questions/19848055/installing-curl-to-php-cli/29656078#29656078

and some information about PHP versions and dates is here: http://stackoverflow.com/a/15171355

Thanks!
